### ISSUE TITLE
ClassCastException on models with multiple UsageScenanios with different types of Workload, too many new Users with mulitple openWL Scenarios

### ISSUE THREAD

**Issue by stiesssh** [PalladioSimulator/Palladio-Analyzer-Slingshot-Extension-PCM-Core#21]
## Problem Description (1)
* Used Model : Mosaic RMCU, with its three usage scenarios (1 openWL, 2 closedWL)
* Exception, once the first users finish:
  ```
  java.lang.ClassCastException: class org.palladiosimulator.pcm.usagemodel.impl.ClosedWorkloadImpl cannot be cast to class org.palladiosimulator.pcm.usagemodel.OpenWorkload (org.palladiosimulator.pcm.usagemodel.impl.ClosedWorkloadImpl and org.palladiosimulator.pcm.usagemodel.OpenWorkload are in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @6234237b)
	at org.palladiosimulator.analyzer.slingshot.behavior.usagesimulation.UsageSimulationBehavior.interpreteOpenWorkload(UsageSimulationBehavior.java:182)
	at org.palladiosimulator.analyzer.slingshot.behavior.usagesimulation.UsageSimulationBehavior.onInterArrivalUserInitiated(UsageSimulationBehavior.java:290)
  [...]
  ```

## Problem Description (2)
* Used Model : Any Usage model with at least two open workload scenarios
* for each starting user, the simulation initiate a new user for each scenario, thereby exponentially increasing the number of users.  

## Problem Cause
* when a new openWL user enters the system (not at simulation start, but the first users after that) the operation `onInterArrivalUserInitiated` gets *all* `UsageScenarioInterpretationContext` and tries to interprete them as openWL, eventhoug some context have closed WL (1) and eventhough all but one are not related to the current user (2).

**Comment by stiesssh** [comment]
excerpt from log: 
```
[...]
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 0.0: UserStarted(UserStarted-518F8CC)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 0.0: UserStarted(UserStarted-25CC4EC6)
[...]
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 1.0: UserStarted(UserStarted-4F59DE3)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 1.0: UserStarted(UserStarted-2062E0C7)
[...]
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 2.0: UserStarted(UserStarted-5D7DFC92)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 2.0: UsageModelPassedElement(UsageModelPassedElement-43CD86CB)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 2.0: UsageModelPassedElement(UsageModelPassedElement-6E03B010)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 2.0: UserStarted(UserStarted-57014154)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 2.0: UserStarted(UserStarted-22DCF131)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[...]

[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UserStarted(UserStarted-7BCFAF00)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UsageModelPassedElement(UsageModelPassedElement-43CD86CB)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UserStarted(UserStarted-45FE1E98)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UsageModelPassedElement(UsageModelPassedElement-43CD86CB)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UserStarted(UserStarted-539F1A3D)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UsageModelPassedElement(UsageModelPassedElement-6E03B010)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UserStarted(UserStarted-4670CBA6)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UserStarted(UserStarted-2213B2C1)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UsageModelPassedElement(UsageModelPassedElement-43CD86CB)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UsageModelPassedElement(UsageModelPassedElement-6E03B010)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[Worker-1: Launching minimal fcfs open] INFO : Even dispatched at 3.0: UserStarted(UserStarted-5536A33E)
[Worker-1: Launching minimal fcfs open] INFO : Post interception result was successful: true
[...]
```

**Comment by stiesssh** [comment]
closed by PR #23
